org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/launch/internal/DefaultExternalLaunchConfigurationManager.java

```diff
@@ -170,14 +170,26 @@ private void updateLaunchConfiguration(ILaunchConfigurationWorkingCopy configura
      */
     private class LaunchConfigurationListener implements ILaunchConfigurationListener, EventListener {
 
+        // If another ILaunchConfigurationListener changes the target launch configuration then
+        // a new change event is generated. That event is handled synchronously invoking this
+        // listener again and causing a stack overflow error.
+        // see https://github.com/eclipse/buildship/issues/617
+        private boolean configChangeCalled;
+
         @Override
         public void launchConfigurationAdded(ILaunchConfiguration configuration) {
             updateClasspathProvider(configuration);
         }
 
         @Override
         public void launchConfigurationChanged(ILaunchConfiguration configuration) {
-            updateClasspathProvider(configuration);
+            if (this.configChangeCalled) {
+                return;
+            } else {
+                this.configChangeCalled = true;
+                updateClasspathProvider(configuration);
+                this.configChangeCalled = false;
```

Comment:
Needs to be in a finally block. Otherwise any error will put us into noop mode forever.

Reply:
Fixed.